catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/SeedCommand.java

```diff
@@ -0,0 +1,241 @@
+import static ddf.catalog.filter.FilterDelegate.WILDCARD_CHAR;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+
+import org.apache.karaf.shell.api.action.Action;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.Option;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.geotools.filter.text.cql2.CQL;
+import org.opengis.filter.Filter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.operation.impl.ResourceRequestById;
+import ddf.catalog.resource.ResourceNotFoundException;
+import ddf.catalog.resource.ResourceNotSupportedException;
+
+//@formatter:off
+@Service
+@Command(
+        scope = CatalogCommands.NAMESPACE,
+        name = "seed",
+        description = "Seeds the local metacard and resource caches from the enterprise or from specific federated sources.",
+        detailedDescription = "This command will query the enterprise or the specified sources for metacards in increments "
+                + "the size of the `--resource-limit` argument (default 20) until that number of resource downloads "
+                + "have started or until there are no more results. Local resources will not be added to the cache. "
+                + "This command will not re-download resources that are up-to-date in the cache, so subsequent runs "
+                + "of the command will attempt to cache metacards and resources that have not already been cached.\n\n"
+                + "Note: this command will trigger resource downloads in the background and they may continue after "
+                + "control is returned to the console. Also, resource caching must be enabled in the catalog framework "
+                + "for this command to seed the resource cache.")
+//@formatter:on
+public class SeedCommand extends SubjectCommands implements Action {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SeedCommand.class);
+
+    private static final Map<String, Serializable> CACHE_UPDATE_PROPERTIES =
+            Collections.singletonMap("mode", "update");
+
+    private static final String RESOURCE_CACHE_STATUS = "internal.local-resource";
```

Comment:
Since both this and the catalog-core-resourcestatusplugin use this constant, maybe it should be moved to a common area.  If that constant changes in the plugin, then this command breaks.